x-pack/plugin/esql/qa/server/src/main/java/org/elasticsearch/xpack/esql/qa/rest/RestEsqlTestCase.java

```diff
@@ -295,6 +374,105 @@ public void testCSVNoHeaderMode() throws IOException {
         assertEquals("keyword0,0\r\n", actual);
     }
 
+    public void testOutOfRangeComparisons() throws IOException {
+        final int NUM_SINGLE_VALUE_ROWS = 100;
+        bulkLoadTestData(NUM_SINGLE_VALUE_ROWS);
+        bulkLoadTestData(10, NUM_SINGLE_VALUE_ROWS, false, RestEsqlTestCase::createDocumentWithMVs);
+        bulkLoadTestData(5, NUM_SINGLE_VALUE_ROWS + 10, false, RestEsqlTestCase::createDocumentWithNulls);
+
+        List<String> dataTypes = List.of(
+            "alias_integer",
+            "byte",
+            "short",
+            "integer",
+            "long",
+            // TODO: https://github.com/elastic/elasticsearch/issues/102935
+            // "unsigned_long",
+            // TODO: https://github.com/elastic/elasticsearch/issues/100130
+            // "half_float",
+            // "float",
+            "double",
+            "scaled_float"
+        );
+
+        List<String> trueForSingleValuesComparisons = List.of(
+            " < " + HUMONGOUS_DOUBLE,
+            " <= " + HUMONGOUS_DOUBLE,
+            " > -" + HUMONGOUS_DOUBLE,
+            " >= -" + HUMONGOUS_DOUBLE,
+            " != " + HUMONGOUS_DOUBLE,
+            " != -" + HUMONGOUS_DOUBLE,
+            " < " + INFTY,
+            " <= " + INFTY,
+            " > -" + INFTY,
+            " >= -" + INFTY,
+            " != " + INFTY,
+            " != -" + INFTY,
+            " != " + NAN
+
+        );
+        List<String> alwaysFalseComparisons = List.of(
+            " < -" + HUMONGOUS_DOUBLE,
+            " <= -" + HUMONGOUS_DOUBLE,
+            " > " + HUMONGOUS_DOUBLE,
+            " >= " + HUMONGOUS_DOUBLE,
+            " == " + HUMONGOUS_DOUBLE,
+            " == -" + HUMONGOUS_DOUBLE,
+            " < -" + INFTY,
+            " <= -" + INFTY,
+            " > " + INFTY,
+            " >= " + INFTY,
+            " == " + INFTY,
+            " == -" + INFTY,
+            " < " + NAN,
+            " <= " + NAN,
+            " > " + NAN,
+            " >= " + NAN,
+            " == " + NAN,
+            " < to_double(null)",
+            " <= to_double(null)",
+            " > to_double(null)",
+            " >= to_double(null)",
+            " == to_double(null)",
+            " != to_double(null)"
+        );
+
+        for (String fieldWithType : dataTypes) {
+            for (String trueComparison : trueForSingleValuesComparisons) {
+                var query = builder().query(
+                    LoggerMessageFormat.format(null, "from {} | where {} {}", testIndexName(), fieldWithType, trueComparison)
+                );
+                var result = runEsql(query);
```

Comment:
There are quite a lot of queries this method is testing, around 240.
I would consider introducing some randomization and slim them down a bit.

For example, I would consider the values in `trueForSingleValuesComparisons` in three categories: HUMONGOUS_DOUBLE, INFINITY and NAN. I'd randomly choose one comparison with HUMONGOUS_DOUBLE, one with INFINITY and one with the NAN. With `alwaysFalseComparisons` I'd do something similar, but split the values in four categories: HUMONGOUS_DOUBLE, INFINITY, NAN, to_double and choose randomly one value from each.

Reply:
I slimmed it down to 66 queries (88 once `half_float` and `float` are enabled).

I wouldn't slim it down any more, the slimmer version already doesn't hit all code paths on every test execution, which could lead to regressions making it into main.